### PR TITLE
dispatch-build-bottle: run `brew update`

### PR DESCRIPTION
**Pull request by carlocab**
Still seeing duplicate bottle jobs: https://github.com/Homebrew/homebrew-core/actions/runs/10820149397

### CODE REVIEW

**Comment by Bo98** [comment]
That'll be because it was dispatched in the 5 minute window that a PR was open.

**Comment by Bo98** [comment]
Could also I guess make the upload workflow wait for the PR to be merged and thus hold on to the concurrency lock.

**Review by Bo98 — APPROVED** [review]
I do agree with this in general though - just wouldn't quite catch the example you linked and may need to be paired with one of the above.

File: .github/workflows/dispatch-build-bottle.yml [hunk 1]
```diff
@@ -90,6 +90,10 @@ jobs:
                 return {runner: s, cleanup: true};
             });
 
+      - name: Ensure homebrew/core is up-to-date
+        if: github.ref == 'master'
```
Comment (ZhongRuoyu) [inline .github/workflows/dispatch-build-bottle.yml:94]:
```suggestion
        if: github.ref == 'ref/heads/master'
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by carlocab** [comment]
> I do agree with this in general though - just wouldn't quite catch the example you linked and may need to be paired with one of the above.

Done the latter.

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**